### PR TITLE
add notification to admin dash if no stats generated yet 

### LINKED ISSUE #344: getDailyStats `HttpError` being thrown with invalid status code
in [analytics/operations.ts#LN27](https://github.com/wasp-lang/open-saas/blob/6e41a1032a95365fee835dc5e02ea433fd986ba7/template/app/src/analytics/operations.ts#L27) there is a `throw new HttpError(204, 'No daily stats generated yet.');`

When deployed, this throws repeatedly:
```
Error: statusCode has to be integer in range [400, 600).
    at HttpError (/app/.wasp/out/sdk/wasp/server/HttpError.ts:15:13)
    at getDailyStats$2 (/app/src/analytics/operations.ts:27:11)
    at <anonymous> (/app/.wasp/build/server/src/middleware/operations.ts:14:24)
    at <anonymous> (/app/.wasp/out/sdk/wasp/server/utils.ts:26:5)
```

Based on the error, it appears `HttpError` in Wasp enforces a status code >= 400 && <= 600:
https://github.com/wasp-lang/wasp/blob/f2b966ad44012809b3141b2083e2368c31ed0bb1/waspc/data/Generator/templates/sdk/wasp/server/HttpError.ts#L14-L16

### PR DESCRIPTION
**Pull request by vincanger**
## Description

> Fixes #344 and removes error being thrown if no stats have been calculated and generated yet by the `dailyStatsJob`

## Contributor Checklist

> Make sure to do the following steps if they are applicable to your PR:

- [ ] **Update e2e tests**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/template/e2e-tests](/template/e2e-tests) also.
- [x] **Update demo app**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/opensaas-sh/app_diff](/opensaas-sh/app_diff) also. Check [/opensaas-sh/README.md](/opensaas-sh/README.md) for details.
- [ ] **Update docs**: If needed, update the [/opensaas-sh/blog/src/content/docs](/opensaas-sh/blog/src/content/docs).

### CODE REVIEW

File: template/app/src/admin/dashboards/analytics/AnalyticsDashboardPage.tsx [hunk 1]
```diff
@@ -7,31 +7,52 @@ import TotalRevenueCard from './TotalRevenueCard';
...
-        <TotalPageViewsCard
-          totalPageViews={stats?.dailyStats.totalViews}
-          prevDayViewsChangePercent={stats?.dailyStats.prevDayViewsChangePercent}
-        />
-        <TotalRevenueCard dailyStats={stats?.dailyStats} weeklyStats={stats?.weeklyStats} isLoading={isLoading} />
-        <TotalPayingUsersCard dailyStats={stats?.dailyStats} isLoading={isLoading} />
-        <TotalSignupsCard dailyStats={stats?.dailyStats} isLoading={isLoading} />
-      </div>
+      <div className='relative'>
+        <div className={`${!stats ? 'opacity-25' : ''}`}>
```
Comment (infomiho) [inline template/app/src/admin/dashboards/analytics/AnalyticsDashboardPage.tsx:20]: I'd recommend using [`clsx`](https://www.npmjs.com/package/clsx) (and maybe [`twMerge`](https://www.npmjs.com/package/tailwind-merge) alongside it) so you can write this more easily:
```ts
<div className={cn({
  opacity-25: !stats,
})}>
...
</div>
```

This enables you to compose styles in an object and then the `clsx` gives you the final string. It's easier to write and read IMHO.

ShadCN packs it into a `cn` helper fn:
```ts
import { clsx, type ClassValue } from "clsx"
import { twMerge } from "tailwind-merge"

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs))
}
```

**Review by infomiho — COMMENTED** [review]
Looking good! Left one comment.

Also, I'd hide the banner below until we have the stats:

Screenshot 2024-12-17 at 16 29 10

File: template/app/src/admin/dashboards/analytics/AnalyticsDashboardPage.tsx (hunk 1 above)
Comment (vincanger) [reply template/app/src/admin/dashboards/analytics/AnalyticsDashboardPage.tsx:20]: we actually alread have the cn helper function i just forgot to use it :)

**Comment by vincanger** [comment]
@infomiho no I just left it becuase its only for opensaas.sh and for that we have a lot stats data in the db

**Comment by infomiho** [comment]
@vincanger Makes sense, but it still feels little dirty not to hide it _in case_ we don't have any data. I know we have it, but you know, to make this as nice as possible of an implementation :)

**Comment by vincanger** [comment]
ok I fixed it and pushed to `main`

**Pull request merged**